### PR TITLE
gha: Remove unneeded strategy on workflow file

### LINKED ISSUE #3275: Error inspektor-gadget.yml
## Description

Encountering an error "Missing property 'matrix'" within the GitHub Actions YAML configuration file. Specifically, the error occurs under the "strategy: fail-fast: false" section, indicating a potential issue with the nesting or structure of the matrix property.

## Impact

The workflow fails to execute properly due to the missing matrix property, preventing the CI/CD pipeline from running the specified jobs.

## Environment and steps to reproduce

- Go to .github\workflows\inspektor-gadget.yml

## Expected behavior

The workflow should correctly recognize and utilize the matrix property within the strategy section.

### PR DESCRIPTION
**Pull request by andoriyaprashant**
## Title: describe the change in one sentence

This PR removes the `strategy` section from the `.github/workflows/inspektor-gadget.yml` file. The `strategy: fail-fast: false` lines were causing issues with the workflow execution, specifically triggering a "Missing property 'matrix'" error.

## How to use

Review the changes in the .github/workflows/inspektor-gadget.yml file.

### CODE REVIEW

File: .github/workflows/inspektor-gadget.yml [hunk 1]
```diff
@@ -732,8 +732,6 @@ jobs:
       # in case of public fork repo/packages permissions will always be read
       contents: read
       packages: write
-    strategy:
-      fail-fast: false
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply .github/workflows/inspektor-gadget.yml:736]: The issue is that there is not any matrix defined there. I think https://github.com/inspektor-gadget/inspektor-gadget/commit/77582cbbf31d20b3f62e0ed00d3aea621637395f forgot to remove the whole strategy.

**Comment by mauriciovasquezbernal** [comment]
Please fix the commit title "inspekto-gadget.yml file fixed". 

nit: It's better to use an imperative mood when writing commit titles / messages:

Inspektor-gadget.yml file fixed -> Fix inspektor-gadget.yaml file 

Also, it's good to be specific on the commit title:

Inspektor-gadget.yml file fixed > gha: Remove uneeded strategy on workflow file

(gha stands for github actions), so by using this prefix you make it clear what par of the project is your change about).

**Comment by andoriyaprashant** [comment]
Sir @mauriciovasquezbernal @eiffel-fl  I have made all the modifications you suggested. Please check them, and if there are any additional changes needed, kindly let me know.  I will try to make those changes. Thank you

**Comment by andoriyaprashant** [comment]
@mauriciovasquezbernal @eiffel-fl   Sir, I used the wrong command, which caused some issues in this pull request. Therefore, I have opened a new pull request. Please check it #3302

**Pull request closed without merging**